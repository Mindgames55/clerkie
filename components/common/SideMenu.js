import styles from './SideMenu.module.css'
import MenuItem from './MenuItem';
import HomeIcon from '../../public/home.svg'
import FriendsIcon from '../../public/friends.svg'
import Logo from '../../public/logo.svg'

const homeItem = {
    path: '/',
    title: 'Home'
}

const friendsItem = {
    path: 'friends',
    title: 'Friends'
}

const menuItems = [homeItem, friendsItem];

export default function SideMenu() {
    return (
        <div className={styles.container}>
            <div className={styles.logo}>
                <Logo />
                <span className={styles['logo-title']}>Clerkie Challenge</span>
            </div>
            <nav>
                <ul>
                    {menuItems.map(item => 
                        (
                            <MenuItem key={item.path} href={item.path} title={item.title}>
                                {item.path === '/' ? <HomeIcon /> : <FriendsIcon />}
                            </MenuItem>
                        )
                    )}
                </ul>
            </nav>
        </div>
    );
}