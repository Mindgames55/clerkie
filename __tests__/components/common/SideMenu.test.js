import { render } from '@testing-library/react';
import SideMenu from '@/components/common/SideMenu';

jest.mock('next/navigation', () => ({
    useSelectedLayoutSegment: jest.fn(), // return undefined for base route
}))

it('renders SideMenu correctly', () => {
  const { container } = render(<SideMenu />);
  expect(container).toMatchSnapshot();
});