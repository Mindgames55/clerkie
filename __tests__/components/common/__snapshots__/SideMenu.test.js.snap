// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SideMenu correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="logo"
    >
      <span />
      <span
        class="logo-title"
      >
        Clerkie Challenge
      </span>
    </div>
    <nav>
      <ul>
        <li>
          <a
            class="item-active"
            href="/"
          >
            <span />
            <span
              class="text"
            >
              Home
            </span>
          </a>
        </li>
        <li>
          <a
            class="item"
            href="friends"
          >
            <span />
            <span
              class="text"
            >
              Friends
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
